src/arch_ppc.py

```diff
@@ -173,6 +178,87 @@ def match(self, matcher: AsmMatcher) -> Optional[Replacement]:
         return None
 
 
+class FloatishToUintPattern(SimpleAsmPattern):
+    pattern = make_pattern("bl __cvt_fp2unsigned")
+
+    def replace(self, m: AsmMatch) -> Optional[Replacement]:
+        return Replacement(
+            [AsmInstruction("cvt.u.d.fictive", [Register("r3"), Register("f1")])],
+            len(m.body),
+        )
+
+
+class FloatishToSintIrPattern(IrPattern):
+    # This pattern handles converting either f32 or f64 into a signed int
+    # The `fctiwz` instruction does all the work; this pattern is just to
+    # elide the stack store/load pair.
+    replacement = "fctiwz.fictive $i, $f"
+    parts = [
+        "fctiwz $t, $f",
+        "stfd $t, (N-4)($r1)",
+        "lwz $i, N($r1)",
+    ]
+
+
+class CheckConstantMixin:
+    def check(self, m: IrMatch) -> bool:
+        # TODO: Also validate that `K($k)` is the expected constant in rodata
+        return m.symbolic_registers["k"] in (Register("r2"), Register("r13"))
+
+
+class SintToDoubleIrPattern(IrPattern, CheckConstantMixin):
+    # The replacement asm for these patterns reference the float constant `K($k)`
+    # as an input, even though the value is ignored. This is needed to mark `$k`
+    # as an input to the pattern for matching.
```

Comment:
Yeah - my reasoning was that it would probably be a mistake to omit an input here? It seemed like a reasonable assertion, but I'd be fine with the change you proposed if there were an explicit use case for it.